packages/react/src/ErrorBoundary.tsx

```diff
@@ -33,11 +33,10 @@ export interface ErrorBoundaryFallbackProps<TError extends Error = Error> {
   reset: () => void
 }
 
-type EnabledCallback = (error: Error) => boolean
-type Enabled = ConstructorType<Error> | EnabledCallback
-const shouldCatch = (error: Error, enabled: Enabled) =>
-  (enabled.prototype instanceof Error && error instanceof enabled) ||
-  (typeof enabled === 'function' && (enabled as EnabledCallback)(error))
+type ShouldCatchCallback = (error: Error) => boolean
+type ShouldCatch = ConstructorType<Error> | ShouldCatchCallback
+const checkErrorBoundary = (shouldCatch: ShouldCatch, error: Error) =>
+  shouldCatch.prototype instanceof Error ? error instanceof shouldCatch : (shouldCatch as ShouldCatchCallback)(error)
```

Comment:
I checked if shouldCatch(ErrorConstructor or ShouldCatchCallback) extend Error correctly. after it enabled will be EnabledCallback.